dynamodb/src/main/java/com/yahoo/ycsb/db/DynamoDBClient.java

```diff
@@ -337,20 +333,20 @@ public Status delete(String table, String key) {
         HashMap<String, ByteIterator> rItems = new HashMap<String, ByteIterator>(item.size());
 
         for (Entry<String, AttributeValue> attr : item.entrySet()) {
-            logger.debug(String.format("Result- key: %s, value: %s", attr.getKey(), attr.getValue()) );
+            logger.debug(String.format("Result- key: %s, value: %s", attr.getKey(), attr.getValue()));
             rItems.put(attr.getKey(), new StringByteIterator(attr.getValue().getS()));
         }
         return rItems;
     }
 
-    private Key createPrimaryKey(String key) {
-        Key k;
+    private Map<String, AttributeValue> createPrimaryKey(String key) {
+        Map<String, AttributeValue> k = new HashMap<String, AttributeValue>();
         if (primaryKeyType == PrimaryKeyType.HASH) {
-          k = new Key().withHashKeyElement(new AttributeValue().withS(key));
+          k.put(primaryKeyName, new AttributeValue().withS(key));
         } else if (primaryKeyType == PrimaryKeyType.HASH_AND_RANGE) {
-          k = new Key()
-                .withHashKeyElement(new AttributeValue().withS(hashKeyValue))
-                .withRangeKeyElement(new AttributeValue().withS(key));
+          k = new HashMap<String, AttributeValue>();
```

Comment:
Does k need to be a new hashmap here? Looks like it is initialized on line 343 already.